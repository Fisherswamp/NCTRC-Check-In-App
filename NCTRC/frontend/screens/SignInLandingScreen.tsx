import React from 'react';
import { SafeAreaView, StyleSheet, Text, View, Button, Image, TouchableOpacity } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';
import { AuthStackParamList, AppScreens } from '../Index';
import { styles } from './Styles';

type SingInLandingScreenNavigationProps = StackNavigationProp<AuthStackParamList, AppScreens.SignInLanding>;

interface SignInLandingScreenProps {
    navigation: SingInLandingScreenNavigationProps;
}

const SignInLandingScreen: React.FunctionComponent<SignInLandingScreenProps> = (props) => {
    let { navigation } = props;
    return (
        <SafeAreaView style={styles.container}>
             <TouchableOpacity style={styles.backButton} onPress={() => navigation.pop()}>
                     <Image source={require('./../assets/backbutton.png')} style={{ width: 75, height: 75 }}></Image>
                </TouchableOpacity> 
                <TouchableOpacity style={styles.logo} onPress={() => navigation.popToTop()}>
                     <Image source={require('./../assets/NCTRClogo.png')} style={{ width: 150, height: 150 }}></Image>
                </TouchableOpacity> 
            <View style={styles.homeContainer}>
            <View style={styles.landingScreenBoxes}>
                <Text style={styles.landingScreenBoxText}>Welcome to NCTRC!</Text>
                <TouchableOpacity style={styles.smallButton}onPress={() => navigation.popToTop()}>
                <Text style={styles.buttonText}>
                Home
                 </Text>  
                </TouchableOpacity>
                </View>
            </View>
        </SafeAreaView>
    );
};
export default SignInLandingScreen;
