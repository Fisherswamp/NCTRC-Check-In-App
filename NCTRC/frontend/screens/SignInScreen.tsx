import React from 'react';
import { SafeAreaView, StyleSheet, Text, View, Button, Image } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';
import { AuthStackParamList, AppScreens } from '../index';
type SingInScreenNavigationProps = StackNavigationProp<AuthStackParamList, AppScreens.SignIn>;
interface SignInScreenProps {
    navigation: SingInScreenNavigationProps;
}
const styles = StyleSheet.create({

});
const SignInScreen: React.FunctionComponent<SignInScreenProps> = (props) => {
    const { navigation } = props;
return (
        <SafeAreaView>
            <View>
                <Image source={require('./../assets/NCTRClogo.png')} style={{width: 400, height: 400}}></Image>
                <Text>SignInScreen</Text>
                <Button color='#884633' title="Home" onPress={() => navigation.navigate(AppScreens.Home)} />

            </View>
        </SafeAreaView>
    );
};
export default SignInScreen;