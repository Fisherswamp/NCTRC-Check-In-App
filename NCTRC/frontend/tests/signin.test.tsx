
import React from "react";
import  SignInScreen from '../screens/SignInScreen';
import  SignInLandingScreen from '../screens/SignInLandingScreen';
import { shallow, configure } from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const props = {
    navigation: {
        addListener: jest.fn(), 
        canGoBack: jest.fn(), 
        dangerouslyGetParent: jest.fn(), 
        dangerouslyGetState: jest.fn(), 
        dispatch: jest.fn(), 
        goBack: jest.fn(), 
        isFocused: jest.fn(), 
        navigate: jest.fn(), 
        pop: jest.fn(), 
        popToTop: jest.fn(), 
        push: jest.fn(), 
        removeListener: jest.fn(), 
        replace: jest.fn(), 
        reset: jest.fn(), 
        setOptions: jest.fn(), 
        setParams: jest.fn(), 
    }, 
}

describe('SignInScreenLanding renders', () => {
    it('App render', async() => {
      const tree = renderer.create(<SignInLandingScreen navigation={props.navigation} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

describe('SignInScreen renders', () => {
    it('App render', async() => {
      const tree = renderer.create(<SignInScreen navigation={props.navigation} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});


describe('SignInScreen elements', () => {
    it("SignInScreen elements", async() => {
       let wrapper =  shallow(<SignInScreen navigation={props.navigation} />);
       expect(wrapper.find('Image').length).toBe(2); 
       expect(wrapper.find('Text').length).toBe(1); 
       expect(wrapper.find('View').length).toBe(1); 

    })
   
});

describe('SignInLanding elements', () => {
    it("SignInLanding elements", async() => {
       let wrapper =  shallow(<SignInLandingScreen navigation={props.navigation} />);
       expect(wrapper.find('Image').length).toBe(2); 
       expect(wrapper.find('Text').length).toBe(2); 
       expect(wrapper.find('View').length).toBe(2); 

    })
   
});
jest.useFakeTimers(); 