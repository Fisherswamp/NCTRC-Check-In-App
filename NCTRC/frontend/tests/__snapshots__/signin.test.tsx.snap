// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen renders App render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/backbutton.png",
        }
      }
      style={
        Object {
          "height": 75,
          "width": 75,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "marginTop": -40,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/NCTRClogo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="First Name"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#884633",
          "borderRadius": 9,
          "borderWidth": 1,
          "height": 60,
          "margin": 10,
          "paddingHorizontal": 10,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Last Name"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#884633",
          "borderRadius": 9,
          "borderWidth": 1,
          "height": 60,
          "margin": 10,
          "paddingHorizontal": 10,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email Address"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "#884633",
          "borderRadius": 9,
          "borderWidth": 1,
          "height": 60,
          "margin": 10,
          "paddingHorizontal": 10,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 20,
          "marginBottom": 50,
          "opacity": 1,
          "width": 180,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Sign In
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`SignInScreenLanding renders App render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/backbutton.png",
        }
      }
      style={
        Object {
          "height": 75,
          "width": 75,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "marginTop": -40,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/NCTRClogo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#884633",
          "borderRadius": 9,
          "borderWidth": 1,
          "marginTop": -150,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#884633",
            "fontSize": 15,
            "margin": 40,
          }
        }
      >
        Welcome to NCTRC!
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#884633",
            "borderRadius": 9,
            "height": 60,
            "margin": 20,
            "marginBottom": 50,
            "opacity": 1,
            "width": 180,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "marginTop": 18,
            }
          }
        >
          Home
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
